invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/DocumentDetails/components/DocumentAccessRestrictions/state/reducer.js

```diff
@@ -0,0 +1,56 @@
+import {
+  IS_LOADING,
+  SUCCESS,
+  HAS_ERROR,
+  SET_RESTRICTIONS_IS_LOADING,
+  SET_RESTRICTIONS_SUCCESS,
+  SET_RESTRICTIONS_HAS_ERROR,
+} from './types';
+
+export const initialState = {
+  isLoading: true,
+  hasError: false,
+  data: {},
+  error: {},
+};
+
+export default (state = initialState, action) => {
+  switch (action.type) {
+    case IS_LOADING:
+      return { ...state, isLoading: true };
```

Comment:
This is what I felt weird because you have 2 action types but they set the same keys in the state. We need to keep one and I proposed to move the `searchPatrons` because I see that we use in the action type string the format `{actionName/state}`.